src/uct/ib/rc/base/rc_ep.h

```diff
@@ -92,26 +92,40 @@ enum {
 /*
  * check for FC credits and add FC protocol bits (if any)
  */
-#define UCT_RC_CHECK_FC_WND(_iface, _ep, _am_id) \
-    do { \
+#define UCT_RC_CHECK_FC_WND(_fc, _stats)\
+    if (_fc.fc_wnd <= 0) { \
```

Comment:
pls wrap _fc in ( ) in both macros, and pass it as a pointer (e.g use (_fc)->fc_wnd)